src/components/Indicator/Indicator.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+
+import {
+  createShorthandFactory,
+  UIComponent,
+  UIComponentProps,
+  commonPropTypes,
+  ColorComponentProps,
+  customPropTypes,
+} from '../../lib'
+import { ReactProps, ShorthandValue } from '../../../types/utils'
+import Icon from '../Icon/Icon'
+
+export interface IndicatorProps extends UIComponentProps, ColorComponentProps {
+  /** The indicator can be direction in different directions. */
+  direction?: 'forward' | 'back' | 'top' | 'bottom'
+
+  /** The indicator can show specific icon if provided. */
+  icon?: ShorthandValue
+}
+
+/**
+ * An indicator is suggest additional content next to the element it is inside.
+ */
+class Indicator extends UIComponent<ReactProps<IndicatorProps>, any> {
```

Comment:
Having concerns regarding the name `Indicator` as it can be anything, not specific to mark the element that can expand. Honestly, hard to find any other name, but rather opening a discussion for others ideas.
The only things that came for now `ExpandIndicator`, `HierarchyIndicator`